### PR TITLE
test(engine): add retry to create job script, in case of leader is not ready

### LINKED ISSUE #7300: engine integration test is not stable because of slow initialization of server master leader
### What did you do?

run engine integration test
ref: https://ci2.pingcap.net/blue/organizations/jenkins/engine_ghpr_integration_test/detail/engine_ghpr_integration_test/1116/pipeline/

detail logs: 
[log-dm_collation-dm_sql_mode.tar.gz](https://github.com/pingcap/tiflow/files/9741937/log-dm_collation-dm_sql_mode.tar.gz)

### What did you expect to see?

_No response_

### What did you see instead?

- create job

```go
[2022/10/09 14:11:09.647 +00:00] [INFO] [server.go:200] [payload="{HidePasswordObject=\"job:{type:DM  config:\\\"task-mode: all\\\\ncollation_compatible: \\\\\\\"strict\\\\\\\"\\\\ntarget-database:\\\\n  host: host.docker.internal\\\\n  port: 4000\\\\n  user: root\\\\n  password: ******\\\\nupstreams:\\\\n  - db-config:\\\\n      host: host.docker.internal\\\\n      port: 3306\\\\n      user: root\\\\n      password: ******\\\\n    source-id: mysql-01\\\\n    block-allow-list: global\\\\n  - db-config:\\\\n      host: host.docker.internal\\\\n      port: 3307\\\\n      user: root\\\\n      password: ******\\\\n    source-id: mysql-02\\\\n    block-allow-list: global\\\\nblock-allow-list:\\\\n  global:\\\\n    do-dbs:\\\\n    - \\\\\\\"sync_collation*\\\\\\\"\\\\n    - \\\\\\\"test_panic\\\\\\\"\\\\n\\\"}  tenant_id:\\\"dm_collation\\\"  project_id:\\\")\\\"\"}"] [request=CreateJob]
```

- create job returns

```json
{"code":2, "message":"[DFLOW:ErrMasterNotInitialized]master is not initialized", "details":[]}
```

- leader initialized time

```go
[2022/10/09 14:11:13.317 +00:00] [INFO] [server.go:927] ["leader is initialized"] [took=4.068551594s]
```

### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

master@https://github.com/pingcap/tiflow/commit/0f2a2539a254bc24fac16f1a73d073f5aa552163

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: close #7300

### What is changed and how it works?

- retry 15 times(30s at most) in create job script, in case of server master leader is not initialized.

#### Tests 

 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/test/utils/create_job [hunk 1]
```diff
@@ -23,5 +23,25 @@ config=$(base64 -i ${job_config} | tr -d \\n)
...
+        break
+    else
+        ((i++))
+        echo -e "create job failed $i-th time, retry later, error: $output" >/dev/stderr
+        sleep 2
+    fi
+done
+
+if [ $i -ge $retry_time ]; then
+	echo "create job failed at last"
```
Comment (lance6716) [inline engine/test/utils/create_job:43]: seems some indent problems

**Review by lance6716 — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**